value-processor/src/org/immutables/value/processor/meta/AttributeBuilderReflection.java

```diff
@@ -339,12 +341,19 @@ private static boolean isPossibleBuildMethod(ValueAttribute valueAttribute, Elem
         return false;
       }
 
+      if (!valueAttribute.containingType.names().possibleAttributeBuilder(possibleBuildMethod.getSimpleName())) {
+        return false;
+      }
+
+      Types typeUtils
+          = valueAttribute.containingType.constitution.protoclass().environment().processing().getTypeUtils();
+
       ExecutableElement candidateBuildMethod = (ExecutableElement) possibleBuildMethod;
       return !candidateBuildMethod.getModifiers().contains(Modifier.STATIC)
           && candidateBuildMethod.getModifiers().contains(Modifier.PUBLIC)
           && candidateBuildMethod.getTypeParameters().isEmpty()
           && candidateBuildMethod.getReturnType().getKind() == TypeKind.DECLARED
-          && (candidateBuildMethod.getReturnType()).equals(valueAttribute.containedTypeElement.asType());
+          && (typeUtils.isSameType(candidateBuildMethod.getReturnType(), valueAttribute.containedTypeElement.asType()));
```

Comment:
I wonder what we've use elsewhere (see `FromSupertypesModel.isEligibleFromType` which might be overkill..) In any case we need to use `CachingElements.getDelegate` before passing these types/elements to any of `Types`/`Elements` util methods as we can have wrapped instances and Javac may throw exception (even if this is not happening for some reason, but for consistency)